src/main/java/com/coherentsolutions/pot/insuranceservice/service/InsurancePackageManagementService.java

```
@@ -64,4 +64,38 @@ public InsurancePackageDto deactivateInsurancePackage(UUID id) {
     return insurancePackageMapper.toInsurancePackageDto(insurancePackage);
   }
 
+  public InsurancePackageDto updateInsurancePackage(UUID id,
+      InsurancePackageDto insurancePackageDto) {
+    InsurancePackage insurancePackage = insurancePackageRepository.findByIdOrThrow(id);
+    if (insurancePackage.getStatus() != PackageStatus.ACTIVE) {
+      insurancePackage.setName(insurancePackageDto.getName());
+      insurancePackage.setStartDate(insurancePackageDto.getStartDate());
+      insurancePackage.setEndDate(insurancePackageDto.getEndDate());
+      insurancePackage.setPayrollFrequency(insurancePackageDto.getPayrollFrequency());
+
+      if (insurancePackageDto.getStatus() == PackageStatus.DEACTIVATED || (
+          insurancePackageDto.getStatus() == null
+              && insurancePackage.getStatus() == PackageStatus.DEACTIVATED)) {
+        insurancePackage.setStatus(PackageStatus.DEACTIVATED);
+      } else {
+        LocalDate now = LocalDate.now();
+        if (now.isBefore(insurancePackage.getStartDate())) {
+          insurancePackage.setStatus(PackageStatus.INITIALIZED);
+        } else if (!now.isAfter(insurancePackage.getEndDate())) {
+          insurancePackage.setStatus(PackageStatus.ACTIVE);
+        } else {
+          insurancePackage.setStatus(PackageStatus.EXPIRED);
+        }
+      }
+
+      insurancePackageRepository.save(insurancePackage);
+      return insurancePackageMapper.toInsurancePackageDto(insurancePackage);
+
+    } else {
+      throw new ResponseStatusException(HttpStatus.BAD_REQUEST,
```

I'd use fail fast approach and simply add the check to the beginning of the function: 
```
if(status == ACTIVE) { throw exc }

// all the code you currently have inside of the if (insurancePackage.getStatus() != PackageStatus.ACTIVE) block
````